### ISSUE TITLE
Support timeout at leaf level

### ISSUE THREAD

**Issue by nomisRev** [kotest/kotest#1551]
**Please describe the feature you'd like to see including any solutions in mind if you have any**

I'm currently using `FreeSpec` [here](https://github.com/arrow-kt/arrow-fx/blob/master/arrow-fx-coroutines/src/test/kotlin/arrow/fx/coroutines/stream/StreamTest.kt#L47), using it to group tests in logical groups.

Since the timeout is applied to the `FreeScope`/test container, in some groups that test concurrent/suspending behavior it times out the group. There is currently no way to apply time-outs differently in `FreeSpec`.

**Comment by sksamuel** [comment]
So one option would be to provide a general timeout for all tests in a spec, then you can override at the leaf level, so like

```kotlin
  var timeout = 1000.seconds
  "test 1" - {
    "test 2".config(timeout = 100.seconds) {
     }
  }
```

Another option is to allow config at every level and again override it where needed

```kotlin
  var defaultTestConfig = TestCaseConfig(timeout = 1000.seconds)
  "test 1" - {
    "test 2".config(timeout = 100.seconds) {
     }
  }
```

**Comment by sksamuel** [comment]
In master, will be included in 4.2.0.
